pandas/io/formats/latex.py

```diff
@@ -284,8 +286,13 @@ def _write_tabular_begin(self, buf, column_format: str):
             <https://en.wikibooks.org/wiki/LaTeX/Tables>`__ e.g 'rcl'
             for 3 columns
         """
-        if self.caption is not None or self.label is not None:
+        if (
+            self.caption is not None
+            or self.label is not None
+            or self.position is not None
+        ):
             # then write output in a nested table/tabular environment
+            buf.write(f"\\begin{{table}}")
```

Comment:
Any obvious reason why this was moved up here? I would find `` buf.write(f"\\begin{{table}}{position_}")`` more readable that the ``write`` below.

Reply:
The reason I did this was only because the line was getting too long, but I can write it as it was for sure. 
Maybe we should unify the way it's done in `_write_tabular_begin` and `_write_longtable_begin` ? But it may be nitpicking

Reply:
> The reason I did this was only because the line was getting too long, but I can write it as it was for sure.

I do find it more readable - but if you want to break the line, you could do so at ``\n``, so after the position.

> Maybe we should unify the way it's done in `_write_tabular_begin` and `_write_longtable_begin` ?

Wow... indeed the difference between `_write_tabular_begin` and `_write_longtable_begin` seems to amount to the environment name: unless I'm mistaken, the merge is definitely worth doing (same for ``_end``). By the way, if the function are unified feel free to move back the ``any([p is not None...`` down there.